### ISSUE TITLE
[java] UnusedAssignment: False Negative related to post-increment(?)

### ISSUE THREAD

**Issue by UncleOwen** [pmd/pmd#6960]
**Affects PMD Version:** 7.27.0-SNAPSHOT

**Rule:** [UnusedAssignment](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#unusedassignment)

**Description:**

This is a follow-up to #6200. While the false-positive was fixed in #6957, the false-negative in the following sample is still there

**Code Sample demonstrating the issue:**

```java
    <test-code>
        <description>blub</description>
        <expected-problems>1</expected-problems>
        <expected-linenumbers>9-9</expected-linenumbers>
        <code><![CDATA[
public class Foo {
    private boolean c;

    private int foo(int x) {
        return x + 1;
    }

    public int visit(int i) {
        int x = c ? foo(i++) : 17;
        return i;
    }
}
        ]]></code>
    </test-code>
```

**Expected outcome:**

PMD should report a violation at line 9, but doesn't. The assignment to `x` is never used. This is a false-negative.

**Running PMD through:** *PMD rule tests*

**Comment by adangel** [comment]
It might be, that it is not reported, because not only the assignment is unused, but the variable is unused as well. The rule has a property "reportUnusedVariables", which by default is disabled.

**Comment by UncleOwen** [comment]
Oh. Guess I should have read the rule's description. Thanks.
